tests/src/main/java/net/neoforged/neoforge/debug/client/ClientTests.java

```diff
@@ -143,6 +146,28 @@ public void renderFirstPersonOverlay(ItemStack stack, EquipmentSlot equipmentSlo
         });
     }
 
+    @TestHolder(description = "Checks existence of enum and texture location", enabledByDefault = true)
+    static void customEquipmentLayerType(final DynamicTest test) {
+        test.eventListeners().forge().addListener((final ClientResourceLoadFinishedEvent event) -> {
+            var layerType = EquipmentClientInfo.LayerType.valueOf("NEOTESTS_LAYER_TYPE");
+            // Check serialized name uses slash
+            if (layerType.getSerializedName().contains(":")) {
+                test.fail(layerType.getSerializedName() + " should not contain a colon as part of its path");
```

Comment:
In contrary to the vanilla gametest system, the test framework does not throw an exception when a test is marked as failed. This means that you have to explicitly return from the test function if there's nothing else that would prevent the code from reaching a subsequent `pass()` call. The second failure case can therefore also not be written in the functional style it currently uses.